node/node.go

```diff
@@ -599,6 +601,10 @@ func (node *Node) Start() error {
 
 	isThisNodeAnExplorerNode := node.NodeConfig.Role() == nodeconfig.ExplorerNode
 
+	if node.NodeConfig.IsOffline {
+		select {} // sleep until program is terminated
```

Comment:
Instead of adding more logic in this already huge function, we can simply leave this function unchanged since we have already changed p2p ip to local address. Otherwise, I would suggest you to write a `runOfflineNode` which is simply `select {}` in the if condition before enter `node.Start()` since `node.Start` is all about p2p pubsub message handling.

Reply:
I'll do the ladder approach then..